extensions/etcdsnapshot/etcdsnapshot.go

```diff
@@ -260,18 +247,38 @@ func RestoreRKE1Snapshot(client *rancher.Client, clusterName string, snapshotRes
 }
 
 // RestoreRKE2K3SSnapshot is a helper function to restore a snapshot on an RKE2 or k3s cluster. Returns error if any.
-func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, initialControlPlaneValue, initialWorkerValue string) error {
-	clusterObject, existingSteveAPIObject, err := clusters.GetProvisioningClusterByName(client, clusterName, fleetNamespace)
+func RestoreRKE2K3SSnapshot(client *rancher.Client, clusterName string, snapshotRestore *rkev1.ETCDSnapshotRestore, restoreSnapshotConfig *apisV1.Cluster) error {
```

Comment:
the new name is a little confusing to me, isn't this just a cluster v1 object? it doesn't represent any part of a snapshot, right?

Reply:
also couldn't we get the clusterName from this object instead of having it as a separate param?

Reply:
Comment above addresses this. As for `clusterName`, you're probably correct. I will try tomorrow.

Reply:
Resolved